### PR TITLE
Rob10470/TZDB_2024b

### PR DESCRIPTION
**Pull request by robert-craig-houston**
https://devtopia.esri.com/runtime/arcade/issues/1425

Updates TZDB to 2024 and fixes an embedding issue caused by missing files.
Resolves HH issue caused by our usage of Posix footers. Last transition MUST be equivalent to the result of the footer at the transition epoch. HH was purging 'duplicate' transitions thus breaking this in several timezones. Previous builds missed this HH issue by pure chance, as we test each known transition period matches in the compiled and non compiled variants of the TZDB.

### CODE REVIEW

File: src/tz.cpp [hunk 1]
```diff
@@ -2128,7 +2116,11 @@ time_zone::init() const
 sys_info
 time_zone::load_sys_info(std::vector<detail::transition>::const_iterator i, sys_seconds tp) const
 {
-    auto load_posix_info = [&](){auto tz = Posix::time_zone{ posix_timezone_ }.get_info(tp); if (tz.begin < transitions_.back().timepoint){tz.begin = transitions_.back().timepoint;} return tz;};
+    auto load_posix_info = [&](){
```
Comment (chri7325) [inline src/tz.cpp]: This just looks like a formatting change. Should this be dropped to avoid changing more than we need for future upgrades?

**Review by chri7325 — APPROVED** [review]
Minor suggestion. No need to re-review

File: src/tz.cpp [hunk 2]
```diff
@@ -2128,7 +2116,7 @@ time_zone::init() const
 sys_info
 time_zone::load_sys_info(std::vector<detail::transition>::const_iterator i, sys_seconds tp) const
 {
-    auto load_posix_info = [&](){auto tz = Posix::time_zone{ posix_timezone_ }.get_info(tp); if (tz.begin < transitions_.back().timepoint){tz.begin = transitions_.back().timepoint;} return tz;};
+    auto load_posix_info = [&](){ auto tz = Posix::time_zone{ posix_timezone_ }.get_info(tp);if (tz.begin < transitions_.back().timepoint){tz.begin = transitions_.back().timepoint;}return tz; };
```
Comment (chri7325) [inline src/tz.cpp]: Almost

Comment (robert-craig-houston) [reply src/tz.cpp]: This isn't code orignal to the library so I was being a little fast and loose with it 👀 

in truth we are likely to need to hand move changes across if we need them, I've checked to the most recent commit and didn't see anything for our usecase.

Comment (chri7325) [reply src/tz.cpp]: Ohh my bad, I thought it was and was trying to leave it untouched

**Review by chri7325 — APPROVED** [review]

**Comment by robert-craig-houston** [comment]
https://runtime-rtc.esri.com/view/vTest/job/vtest/job/3rdparty-interface/1142/

**Comment by robert-craig-houston** [comment]
https://runtime-rtc.esri.com/view/vTest/job/vtest/job/runtimecore-interface/27474/

**Pull request merged**